### DISCUSSION TITLE
请问如何指定特定域名的上游dns服务器

### DISCUSSION THREAD

**Discussion by lilarcor** [IrineSistiana/mosdns#276] [Q&A] [answered]
wiki看了一下，没有找到对应的说明。比如对于xyz.com的域名，想要使用内网dns服务器192.168.1.1来查询，要怎么写呢

**Comment by lilarcor** [comment]
比如在这里能针对xyz.com使用192.168.1.1来查询吗， xyx.com/192.168.1.1
tag: forward_local
type: forward
args:
upstream:
- addr: 222.172.200.68
- addr: 61.166.150.123
- addr: https://223.5.5.5/dns-query

**Comment by IrineSistiana** [comment, marked as answer]
if：
匹配域名
exec: 
转发你想要的上游

**Reply by lilarcor** [reply]
这个语法有点疑惑，域名和上游不是args吗，直接写成这样可以吗？
```
if：
xyz.com
exec:
192.168.1.1
```

**Reply by lilarcor** [reply]
十分感谢，好像可以了，不知道这样写是不是合理
` 
  - tag:                   forward_local
    type:                  fast_forward
    args:
      upstream:
###         **- addr:            192.168.1.1**
        - addr:            ${{ ./dns0.sh }}
        - addr:            ${{ ./dns1.sh }}
   
- tag:                   query_is_local_domain
    type:                  query_matcher
    args:
      domain:
###         **- 'xyz.com'**
        - 'ext:./geosite.dat:cn'`
